src/overlays/actors/ovl_En_Racedog/z_en_racedog.c

```diff
@@ -155,62 +227,66 @@ static InitChainEntry sInitChain[] = {
+void EnRacedog_ChangeAnimation(SkelAnime* skelAnime, AnimationInfoS* animationInfo, s32 index) {
     f32 frameCount;
 
-    arg1 += arg2;
-    if (arg1->frameCount < 0) {
-        frameCount = Animation_GetLastFrame(arg1->animation);
+    animationInfo += index;
+    if (animationInfo->frameCount < 0) {
+        frameCount = Animation_GetLastFrame(animationInfo->animation);
     } else {
-        frameCount = arg1->frameCount;
+        frameCount = animationInfo->frameCount;
     }
 
-    Animation_Change(skelAnime, arg1->animation, arg1->playSpeed + (BREG(88) * 0.1f), arg1->startFrame, frameCount,
-                     arg1->mode, arg1->morphFrames);
+    Animation_Change(skelAnime, animationInfo->animation, animationInfo->playSpeed + (BREG(88) * 0.1f),
+                     animationInfo->startFrame, frameCount, animationInfo->mode, animationInfo->morphFrames);
 }
 
-void func_80B246F4(EnRacedog* this, GlobalContext* globalCtx) {
+void EnRacedog_UpdateCollision(EnRacedog* this, GlobalContext* globalCtx) {
     this->collider.dim.pos.x = this->actor.world.pos.x;
     this->collider.dim.pos.y = this->actor.world.pos.y;
     this->collider.dim.pos.z = this->actor.world.pos.z;
     CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
     Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 26.0f, 10.0f, 0.0f, 5);
 }
 
-s16 func_80B2478C(Path* path, s32 arg1, Vec3f* pos, f32* arg3) {
+/**
+ * Returns the Y-rotation the dog should have to move to the next point along the race track path.
+ * There is a small degree of randomness incorporated into this angle.
+ */
+s16 EnRacedog_GetYRotation(Path* path, s32 pointIndex, Vec3f* pos, f32* distSQ) {
     Vec3s* point;
-    f32 phi_f14;
-    f32 sp1C;
+    f32 xDiffWithRandomDeviation;
+    f32 zDiffWithRandomVariation;
     f32 xDiff;
     f32 zDiff;
 
     if (path != NULL) {
         point = Lib_SegmentedToVirtual(path->points);
-        point = &point[arg1];
-        phi_f14 = (randPlusMinusPoint5Scaled(100.0f) + point->x) - pos->x;
-        sp1C = (randPlusMinusPoint5Scaled(100.0f) + point->z) - pos->z;
+        point = &point[pointIndex];
+        xDiffWithRandomDeviation = (randPlusMinusPoint5Scaled(100.0f) + point->x) - pos->x;
+        zDiffWithRandomVariation = (randPlusMinusPoint5Scaled(100.0f) + point->z) - pos->z;
         xDiff = point->x - pos->x;
         zDiff = point->z - pos->z;
     } else {
-        phi_f14 = 0.0f;
-        sp1C = 0.0f;
+        xDiffWithRandomDeviation = 0.0f;
+        zDiffWithRandomVariation = 0.0f;
         xDiff = 0.0f;
         zDiff = 0.0f;
     }
 
-    *arg3 = SQ(xDiff) + SQ(zDiff);
-    return RADF_TO_BINANG(Math_Acot2F(sp1C, phi_f14));
+    *distSQ = SQ(xDiff) + SQ(zDiff);
+    return RADF_TO_BINANG(Math_Acot2F(zDiffWithRandomVariation, xDiffWithRandomDeviation));
 }
 
-void func_80B248B8(EnRacedog* this, Vec3f* arg1) {
-    f32 sp20;
-    f32 sp1C;
+void EnRacedog_CalculateFloorTangent(EnRacedog* this, Vec3f* floorTangent) {
```

Comment:
This is a strange function, I'm not completely convinced that this is necessarily it does... be nice to have a visualisation of it for various polygons. It's especially odd to multiply the normal components like that, especially with the scalings.

Reply:
The name comes from OoT's Clear_Tag, which does a basically identical calculation:
```c
/**
 * Calculate a floor tangent.
 * This is used for the ground flash display lists and Arwing shadow display lists to snap onto the floor.
 */
void EnClearTag_CalculateFloorTangent(EnClearTag* this) {
    // If there is a floor poly below the Arwing, calculate the floor tangent.
    if (this->actor.floorPoly != NULL) {
        f32 x = COLPOLY_GET_NORMAL(this->actor.floorPoly->normal.x);
        f32 y = COLPOLY_GET_NORMAL(this->actor.floorPoly->normal.y);
        f32 z = COLPOLY_GET_NORMAL(this->actor.floorPoly->normal.z);

        this->floorTangent.x = -Math_FAtan2F(-z * y, 1.0f);
        this->floorTangent.z = Math_FAtan2F(-x * y, 1.0f);
    }
}
```
I asked math-knowers in discord about it and didn't get a major response. So I'm tempted to just leave the name as-is.

Reply:
As per the discussion on discord about this, I changed this to be called `GetFloorRot`, which is similar to how many other functions like this are named in OoT.